Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -1226,40 +1226,50 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
                                 Array.Resize<Collider>(ref colliders, maxQuerySceneResults);
                             }
 
-                            int numColliders = UnityEngine.Physics.OverlapSphereNonAlloc(pointer.Rays[i].Origin, pointer.SphereCastRadius, colliders, ~UnityEngine.Physics.IgnoreRaycastLayer);
-
-                            if (numColliders > 0)
+                            Vector3 testPoint = pointer.Rays[i].Origin;
+                            Vector3 objectHitPoint = testPoint;
+                            GameObject closest = null;
+                            float closestDistance = Mathf.Infinity;
+                            int totalNumColliders = 0;
+
+                            // Go through each layerMask and ensure perform the appropriate OverlapSphereCalculation
+                            // Since this is usually done when a pointer passes a IsInteractionEnabled, maybe we can cache the selected colliders inside the pointer?
+                            foreach (LayerMask layerMask in prioritizedLayerMasks)
```

Comment:
@davidkline-ms this has a side effect of adding N number of spherecasts per frame, where N is the number of prioritized layer masks passed by the caller - I know that some consumers have been seeing some performance issues associated with per-frame raycasts (I think they were fairly expensive).

There's a note on the other place that calls OverlapSphereNonAlloc that it should only get called once per frame, but from this change + the existing code, the number of times I think is actually 3*N (where N = size of array).

Reply:
Ah yeah, in this case it's at-worst N spherecasts per frame, but fewer if we hit something at a higher priority raycast (it returns early in this case). But, the at-worst case is actually probably most of the time, when nothing is being hit.

Reply:
Actually, this will only get hit when a sphere pointer is already near a grabbable object, so it's guaranteed to be near something. QueryScene only runs if interaction is enabled on the pointer, which is only true if the pointer is near a grabbable object.

But still, there is some redundant sphere casting going on, both in this code, and in the code that was there previously. I liked @RogPodge 's suggestion to cache the results from SpherePointer.OnPreSceneQuery.